src/components/DatePicker/index.tsx

```diff
@@ -88,16 +106,30 @@ export const DatePicker = ({
         }}
       />
       <StyledButton onClick={toggle}>
-        <StyledCalendar size={24} color={color.spaceMedium} />
-        <ButtonText hasDate={!!date}>
-          {!!date
-            ? date.toLocaleString(locale, {
-                day: 'numeric',
-                month: 'short',
-                year: 'numeric',
-              })
-            : placeholder}
-        </ButtonText>
+        <div>
+          <StyledCalendar size={24} color={color.spaceMedium} />
+          <ButtonText hasDate={!!date}>
+            {!!date
+              ? date.toLocaleString(locale, {
+                  day: 'numeric',
+                  month: 'short',
+                  year: 'numeric',
+                })
+              : placeholder}
+          </ButtonText>
+        </div>
+        {onReset && date && (
+          <ResetButton
+            onClick={e => {
+              onReset()
+              e.stopPropagation()
+            }}
+            type="button"
+            data-testid="reset-button"
```

Comment:
I'm not sure I see the point, and already in input component the reset-button is also using `data-testid` so I thought for the sake of consistency, i would keep it like that

Reply:
Strong opinion ahead 😅 Forcing ourselves to use accessibility getters for testing ensures we are improving accessibility any time we test something. Besides this by using an accessibility getter instead of `data-testid` we are not tying the tests to implementation details.

Reply:
Fixed!